codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -106,7 +106,7 @@ public void onStreamClosed(Http2Stream stream) {
 
             @Override
             public void onStreamHalfClosed(Http2Stream stream) {
-                if (HALF_CLOSED_LOCAL.equals(stream.state())) {
+                if (HALF_CLOSED_LOCAL == stream.state()) {
```

Comment:
I can also revert this change as its not really related but we should do this anyway as its an enum .

Reply:
no objection to this change here